services/provider/server/server.go

```diff
@@ -1421,7 +1421,7 @@ func (s *OCSProviderServer) appendStorageClassKubeResources(
 					consumerConfig.GetCsiRbdNodeCephUserName(),
 					consumer.Status.Client.OperatorNamespace,
 					kmsServiceName,
-					storageCluster.GetAnnotations()[defaults.KeyRotationEnableAnnotation] == "false",
+					consumer.GetAnnotations()[defaults.KeyRotationEnableAnnotation] == "false",
```

Comment:
why are we comparing to a false value? Is the lack of annotation means true? that does not make common sense.

Reply:
Yes without the disable annotation by default key rotation is true on the CSI side, ref- https://github.com/red-hat-storage/ocs-operator/pull/2817

Reply:
It's a design decision by csi not ocs operator. And as of now it's already documented in this way, If we intend to change that behaviour there has to be a separate discussion on that, that is out of scope for this bug fix.